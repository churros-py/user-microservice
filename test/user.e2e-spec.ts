import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import { print } from 'graphql/language/printer';
import * as request from 'supertest';
import { AppModule } from '../src/app.module';
import { gql } from 'apollo-server-express';

describe('User (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleRef: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleRef.createNestApplication();
    await app.init();
  });

  it('Should return a user', async () => {
    await request(app.getHttpServer())
      .post('/graphql')
      .send({
        query: print(gql`
          query {
            user(userId: "1") {
              id
              email
              name
              isActive
              documentNo
              birthDate
            }
          }
        `),
      })
      .expect(({ body }) => {
        expect(body.errors).toBeUndefined();
        expect(body.data?.user).toMatchSnapshot();
      });
    expect(true).toBe(true);
  });
});
