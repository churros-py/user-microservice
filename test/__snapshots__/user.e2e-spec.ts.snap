// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User (e2e) Should return a user 1`] = `
Object {
  "birthDate": "2022-07-17T21:18:37.789Z",
  "documentNo": "1111111111",
  "email": "example@example.com",
  "id": "1",
  "isActive": true,
  "name": "Test",
}
`;
